src/sql/src/plan/error.rs

```diff
@@ -213,6 +218,31 @@ impl PlanError {
             Self::PostgresConnectionErr { cause } => Some(cause.to_string_with_causes()),
             Self::InvalidProtobufSchema { cause } => Some(cause.to_string_with_causes()),
             Self::InvalidOptionValue { err, .. } => err.detail(),
+            Self::UpsertSinkWithInvalidKey {
+                name,
+                desired_key,
+                valid_keys,
+            } => {
+                let valid_keys = if valid_keys.is_empty() {
+                    "There are no known valid unique keys for the underlying relation.".into()
+                } else {
+                    format!(
```

Comment:
I think this would be interesting to cover, probably in test/testdrive/kafka-avro-upsert-sinks.td. I'll try to do that.